src/cockpit/channels/filesystem.py

```diff
@@ -160,6 +161,46 @@ def do_yield_data(self, options: JsonObject) -> Generator[bytes, None, JsonObjec
             raise ChannelError('internal-error', message=str(exc)) from exc
 
 
+class FSReplaceAttrs:
+    uid: 'int | None' = None
+    gid: 'int | None' = None
+    mode: 'int | None' = None
+    supported_attrs = ['owner', 'group', 'mode']
+
+    def __init__(self, value: JsonObject) -> None:
+        # Any unknown keys throw an error
+        unsupported_attrs = list(value.keys() - self.supported_attrs)
+        if unsupported_attrs:
+            raise ChannelError('protocol-error',
+                               message=f'"attrs" contains unsupported key(s) {unsupported_attrs}',
+                               unsupported_attrs=unsupported_attrs) from None
+
+        self.mode = get_int(value, 'mode', None)
+        owner = get_str_or_int(value, 'owner', None)
+        group = get_str_or_int(value, 'group', None)
+
+        if owner is not None and group is None:
+            raise ChannelError('protocol-error', message='"group" attribute is empty while "owner" is provided')
+        if group is not None and owner is None:
+            raise ChannelError('protocol-error', message='"owner" attribute is empty while "group" is provided')
+
+        if isinstance(owner, str):
```

Comment:
I'm ok with this, I just thought the goal was to mirror fsinfo? I.e. if you want to set an uid, then set `uid` field, not `user`

Also, why is this "owner", not "user"? That's another impedance mismatch between fsinfo and fsreplace.

Reply:
fsinfo can also have `user` as an integer, and I think there's no sense in having both attributes here, so I agree with the way that it's currently implemented in this PR (except that "owner" should indeed be renamed to "user").

Reply:
Fair enough! WFM.

Reply:
Renamed to `user`